In [1]:
import pandas as pd
import numpy as np

# Explore the Data

### Read the Data

In [2]:
crime = pd.read_csv('./crime.csv')
crime

unique_key case_number                  date                  block  \
0         9672482    HX322338  2014-06-28T11:02:00Z     030XX S NORMAL AVE   
1         8652653    HV328417  2012-06-10T12:00:00Z      029XX S LOOMIS ST   
2         9357626    HW500140  2013-10-20T02:00:00Z       029XX S CANAL ST   
3         6043894    HP146278  2008-01-27T06:42:52Z    034XX S PARNELL AVE   
4        12689202    JF230287  2022-05-04T05:05:00Z      029XX S THROOP ST   
...           ...         ...                   ...                    ...   
20077     1807040     G636924  2001-10-23T01:00:00Z  020XX N KILPATRICK AV   
20078     1859444     G701893  2001-11-21T05:00:00Z       014XX N MASON AV   
20079     1353293     G057398  2001-01-28T12:30:00Z     072XX S KIMBARK AV   
20080     1381280     G092820  2001-02-13T07:00:00Z      081XX S VERNON AV   
20081     1606451     G379049  2001-06-29T12:45:00Z     068XX S JEFFERY BL   

       iucr         primary_type  description location_description  arrest  \
0      1310      CRIMINAL DAMAGE  TO PROPERTY            APARTMENT    True   
1      1310      CRIMINAL DAMAGE  TO PROPERTY     RESIDENCE-GARAGE   False   
2      1310      CRIMINAL DAMAGE  TO PROPERTY               STREET   False   
3      1310      CRIMINAL DAMAGE  TO PROPERTY            RESIDENCE   False   
4      1310      CRIMINAL DAMAGE  TO PROPERTY            RESIDENCE   False   
...     ...                  ...          ...                  ...     ...   
20077  0910  MOTOR VEHICLE THEFT   AUTOMOBILE               STREET   False   
20078  0910  MOTOR VEHICLE THEFT   AUTOMOBILE               STREET   False   
20079  0910  MOTOR VEHICLE THEFT   AUTOMOBILE               STREET   False   
20080  0910  MOTOR VEHICLE THEFT   AUTOMOBILE               STREET   False   
20081  0910  MOTOR VEHICLE THEFT   AUTOMOBILE               STREET   False   

       domestic  ...  ward  community_area  fbi_code  x_coordinate  \
0         False  ...  11.0            60.0        14     1173453.0   
1         False  ...  11.0            60.0        14     1168580.0   
2         False  ...  11.0            60.0        14     1173766.0   
3         False  ...  11.0            60.0        14     1173190.0   
4         False  ...  11.0            60.0        14     1169930.0   
...         ...  ...   ...             ...       ...           ...   
20077     False  ...   NaN             NaN        07     1144681.0   
20078     False  ...   NaN             NaN        07     1136382.0   
20079     False  ...   NaN             NaN        07     1186170.0   
20080     False  ...   NaN             NaN        07     1180625.0   
20081     False  ...   NaN             NaN        07     1190691.0   

      y_coordinate  year            updated_on   latitude  longitude  \
0        1884797.0  2014  2018-02-10T03:50:01Z  41.839303 -87.639001   
1        1885324.0  2012  2016-02-04T06:33:39Z  41.840856 -87.656867   
2        1885503.0  2013  2018-02-10T03:50:01Z  41.841234 -87.637831   
3        1882150.0  2008  2018-02-28T03:56:25Z  41.832045 -87.640044   
4        1885355.0  2022  2022-05-11T04:49:57Z  41.840912 -87.651912   
...            ...   ...                   ...        ...        ...   
20077    1913196.0  2001  2015-08-17T03:03:40Z  41.917823 -87.743866   
20078    1909102.0  2001  2015-08-17T03:03:40Z  41.906741 -87.774455   
20079    1857257.0  2001  2015-08-17T03:03:40Z  41.763440 -87.593206   
20080    1850906.0  2001  2015-08-17T03:03:40Z  41.746141 -87.613724   
20081    1859729.0  2001  2015-08-17T03:03:40Z  41.770115 -87.576556   

                            location  
0      (41.839303195, -87.639000957)  
1       (41.840856037, -87.65686732)  
2      (41.841233568, -87.637831421)  
3       (41.83204543, -87.640044437)  
4       (41.84091182, -87.651912438)  
...                              ...  
20077  (41.917823214, -87.743866097)  
20078  (41.906741271, -87.774455458)  
20079   (41.76343982, -87.593206253)  
20080  (41.746141111, -87.613724292) 

### Info

In [3]:
crime.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20082 entries, 0 to 20081
Data columns (total 22 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   unique_key            20082 non-null  int64  
 1   case_number           20082 non-null  object 
 2   date                  20082 non-null  object 
 3   block                 20082 non-null  object 
 4   iucr                  20082 non-null  object 
 5   primary_type          20082 non-null  object 
 6   description           20082 non-null  object 
 7   location_description  20072 non-null  object 
 8   arrest                20082 non-null  bool   
 9   domestic              20082 non-null  bool   
 10  beat                  20082 non-null  int64  
 11  district              20082 non-null  int64  
 12  ward                  14048 non-null  float64
 13  community_area        14038 non-null  float64
 14  fbi_code              20082 non-null  object 
 15  x_coordinate       

Note: ward, community_area, x_coordinate, y_coordinate, latitude, longitude, location may have null value

### DF

In [4]:
chicago_crime = pd.DataFrame(crime)
chicago_crime.size, chicago_crime.shape

(441804, (20082, 22))

In [5]:
# Check the columns
chicago_crime.columns

Index(['unique_key', 'case_number', 'date', 'block', 'iucr', 'primary_type',
       'description', 'location_description', 'arrest', 'domestic', 'beat',
       'district', 'ward', 'community_area', 'fbi_code', 'x_coordinate',
       'y_coordinate', 'year', 'updated_on', 'latitude', 'longitude',
       'location'],
      dtype='object')

In [6]:
# check columns datatypes
chicago_crime.dtypes

unique_key                int64
case_number              object
date                     object
block                    object
iucr                     object
primary_type             object
description              object
location_description     object
arrest                     bool
domestic                   bool
beat                      int64
district                  int64
ward                    float64
community_area          float64
fbi_code                 object
x_coordinate            float64
y_coordinate            float64
year                      int64
updated_on               object
latitude                float64
longitude               float64
location                 object
dtype: object

In [7]:
# show the null values
null_values = chicago_crime.loc[chicago_crime.isna().any(axis=1)]
null_values

unique_key case_number                  date                  block  \
205       2876338    HJ541399  2003-08-05T08:45:00Z        021XX W 52ND ST   
217      11476318    JB474220  2008-03-16T12:00:00Z     048XX S SEELEY AVE   
282       6940140    HR320987  2009-05-14T06:05:00Z      054XX S DAMEN AVE   
735       2377111    HH678579  2002-09-28T04:37:00Z        012XX W 49TH ST   
749       2362517    HH662189  2002-09-21T12:15:00Z  016XX W GARFIELD BLVD   
...           ...         ...                   ...                    ...   
20077     1807040     G636924  2001-10-23T01:00:00Z  020XX N KILPATRICK AV   
20078     1859444     G701893  2001-11-21T05:00:00Z       014XX N MASON AV   
20079     1353293     G057398  2001-01-28T12:30:00Z     072XX S KIMBARK AV   
20080     1381280     G092820  2001-02-13T07:00:00Z      081XX S VERNON AV   
20081     1606451     G379049  2001-06-29T12:45:00Z     068XX S JEFFERY BL   

       iucr                primary_type                     description  \
205    501A               OTHER OFFENSE            ANIMAL ABUSE/NEGLECT   
217    1754  OFFENSE INVOLVING CHILDREN  AGG SEX ASSLT OF CHILD FAM MBR   
282    2017                   NARCOTICS              MANU/DELIVER:CRACK   
735    0460                     BATTERY                          SIMPLE   
749    0460                     BATTERY                          SIMPLE   
...     ...                         ...                             ...   
20077  0910         MOTOR VEHICLE THEFT                      AUTOMOBILE   
20078  0910         MOTOR VEHICLE THEFT                      AUTOMOBILE   
20079  0910         MOTOR VEHICLE THEFT                      AUTOMOBILE   
20080  0910         MOTOR VEHICLE THEFT                      AUTOMOBILE   
20081  0910         MOTOR VEHICLE THEFT                      AUTOMOBILE   

                 location_description  arrest  domestic  ...  ward  \
205                         RESIDENCE   False     False  ...  16.0   
217                         RESIDENCE    True      True  ...  15.0   
282    PARKING LOT/GARAGE(NON.RESID.)    True     False  ...  16.0   
735                            STREET   False     False  ...  20.0   
749                          SIDEWALK   False     False  ...  16.0   
...                               ...     ...       ...  ...   ...   
20077                          STREET   False     False  ...   NaN   
20078                          STREET   False     False  ...   NaN   
20079                          STREET   False     False  ...   NaN   
20080                          STREET   False     False  ...   NaN   
20081                          STREET   False     False  ...   NaN   

       community_area  fbi_code  x_coordinate y_coordinate  year  \
205              61.0        26           NaN          NaN  2003   
217              61.0        02           NaN          NaN  2008   
282              61.0        18           NaN          NaN  2009   
735              61.0       08B           NaN          NaN  2002   
749              61.0       08B           NaN          NaN  2002   
...               ...       ...           ...          ...   ...   
20077             NaN        07     1144681.0    1913196.0  2001   
20078             NaN        07     1136382.0    1909102.0  2001   
20079             NaN        07     1186170.0    1857257.0  2001   
20080             NaN        07     1180625.0    1850906.0  2001   
20081             NaN        07     1190691.0    1859729.0  2001   

                 updated_on   latitude  longitude  \
205    2015-08-17T03:03:40Z        NaN        NaN   
217    2019-01-26T04:02:53Z        NaN        NaN   
282    2015-08-17T03:03:40Z        NaN        NaN   
735    2015-08-17T03:03:40Z        NaN        NaN   
749    2015-08-17T03:03:40Z        NaN        NaN   
...                     ...        ...        ...   
20077  2015-08-17T03:03:40Z  41.917823 -87.743866   
20078  2015-08-17T03:03:40Z  41.906741 -87.774455   
20079  2015-08-17T03:03:40Z  41.76

In [8]:
# Working with non null data
data = chicago_crime.dropna()

In [9]:
# Show the crime types
data['primary_type'].value_counts()

BATTERY                              2479
THEFT                                2346
CRIMINAL DAMAGE                      2010
NARCOTICS                            1385
ASSAULT                               958
OTHER OFFENSE                         916
BURGLARY                              799
MOTOR VEHICLE THEFT                   688
DECEPTIVE PRACTICE                    528
ROBBERY                               479
CRIMINAL TRESPASS                     405
WEAPONS VIOLATION                     240
OFFENSE INVOLVING CHILDREN            142
PUBLIC PEACE VIOLATION                115
PROSTITUTION                          101
SEX OFFENSE                            51
INTERFERENCE WITH PUBLIC OFFICER       48
ARSON                                  37
CRIM SEXUAL ASSAULT                    35
GAMBLING                               30
HOMICIDE                               23
LIQUOR LAW VIOLATION                   20
KIDNAPPING                             18
STALKING                          

In [10]:
# Show crime areas
data['block'].value_counts().head(25)

038XX W ROOSEVELT RD         61
051XX S PULASKI RD           38
032XX W ROOSEVELT RD         35
039XX W ROOSEVELT RD         35
021XX E 87TH ST              35
049XX N MILWAUKEE AVE        32
002XX W GARFIELD BLVD        30
008XX E 103RD ST             29
054XX S WENTWORTH AVE        29
033XX W FILLMORE ST          27
039XX W GRENSHAW ST          26
049XX S ARCHER AVE           26
036XX W DOUGLAS BLVD         26
046XX S DAMEN AVE            25
031XX S ASHLAND AVE          25
051XX S WENTWORTH AVE        24
037XX W ROOSEVELT RD         24
012XX S INDEPENDENCE BLVD    24
038XX W GRENSHAW ST          23
054XX S DAMEN AVE            23
035XX W ROOSEVELT RD         23
045XX S DAMEN AVE            23
053XX S PULASKI RD           23
007XX E 111TH ST             23
013XX S INDEPENDENCE BLVD    22
Name: block, dtype: int64

In [11]:
# I assume some of the least crimes would be severe crimes 
chicago_crime['description'].value_counts().tail(10)

AGGRAVATED - OTHER                             1
EXTORTION                                      1
ARMED WHILE UNDER THE INFLUENCE                1
SEX OFFENDER - FAIL TO REGISTER NEW ADDRESS    1
ATTEMPT ARMED - OTHER DANGEROUS WEAPON         1
POSSESSION: SYNTHETIC MARIJUANA                1
GUN OFFENDER: DUTY TO REGISTER                 1
POSSESS - SYNTHETIC DRUGS                      1
VEHICLE TITLE / REGISTRATION OFFENSE           1
ESCAPE                                         1
Name: description, dtype: int64

In [12]:
chicago_crime['description'].value_counts().size

291

In [ ]:
# check domestic and arrest
# another way: chicago_crime[chicago_crime.columns[8:10]]
chicago_crime.loc[:, ['domestic', 'arrest']]

In [14]:
# Check the arrest total
# chicago_crime['arrest'] == True
count = 0
for i in chicago_crime['arrest']:
    if i == True:
        count +=1
print(count)

5215


In [15]:
# Check the domestic total
# chicago_crime['domestic'] == True
count = 0
for i in chicago_crime['domestic']:
    if i == True:
        count +=1
print(count)

2788


In [16]:
# Store the domestic and arrest columns in a dictionary 
# to use conditionals
# check where arrest and domestic are True
domestic_arrest = {
    'arrest': chicago_crime['arrest'],
    'domestic': chicago_crime['domestic']
}

type(domestic_arrest)


dict

In [ ]:
# show where the arrest and domestic are True
for key, values in domestic_arrest['arrest'].items():
    if key and values == True:
        print(key, values)

In [ ]:
# show where the arrest and domestic are True
for key, values in domestic_arrest['domestic'].items():
    if key and values == True:
        print(key, values)

In [ ]:
# variable to store the index
# where domestic and arrest are True
domestic_arrest_index_true = []

# show the index where domestic and arrest are both True 
for key1, values1 in domestic_arrest['domestic'].items(): # Get the domestic index and value
    if key1 and values1 == True: # if True
        for key2, values2 in domestic_arrest['arrest'].items(): # # Get the arrest index and value
            if key2 and values2 == True: # if True
                if key1 == key2: # both arrest and domestic index must be the same 
                    domestic_arrest_index_true.append(key1) # append the index 
                    print(key1) # index that meets the conditionals
print(domestic_arrest_index_true[:5])

In [20]:
# check the list length
# This is the total where the crime happends domestically and the assialant(s) is/are arrested
len(domestic_arrest_index_true)

527

In [ ]:
# validate the list
count_list = 0

for key1, values1 in domestic_arrest['domestic'].items(): # Get the domestic index and value
    if key1 and values1 == True: # if True
        for key2, values2 in domestic_arrest['arrest'].items(): # # Get the arrest index and value
            if key2 and values2 == True: # if True
                if key1 == key2: # both arrest and domestic index must be the same 
                    count_list += 1
                    print(key1, values1, key2, values2)

In [22]:
# check
print(count_list)

527


Note: list and the check have the same length and values, thus we can use the list (index) for analysis

In [23]:
# Turn the list into pandas Series
index_domestic_arrest = pd.Series(domestic_arrest_index_true)
index_domestic_arrest.head()

0     12
1    103
2    151
3    158
4    160
dtype: int64

Note: I change the list to Series just to ease the analysis.
- Suppose it's easier to do this in pandas

# Variables Navigations

Since I created many variables, this list would help me to navigate throughout the variables for analysis
- data = drop all null values
- chicage_crime = DF for the dataset
- null_values = all rows that possess null values on the dataset
- index_domestic_arrest = Series where domestic and arrest are True
- domestic_arrest_index_true = index_domestic_arrest's List

In [24]:
data.head(2)

unique_key case_number                  date               block  iucr  \
0     9672482    HX322338  2014-06-28T11:02:00Z  030XX S NORMAL AVE  1310   
1     8652653    HV328417  2012-06-10T12:00:00Z   029XX S LOOMIS ST  1310   

      primary_type  description location_description  arrest  domestic  ...  \
0  CRIMINAL DAMAGE  TO PROPERTY            APARTMENT    True     False  ...   
1  CRIMINAL DAMAGE  TO PROPERTY     RESIDENCE-GARAGE   False     False  ...   

   ward  community_area  fbi_code  x_coordinate y_coordinate  year  \
0  11.0            60.0        14     1173453.0    1884797.0  2014   
1  11.0            60.0        14     1168580.0    1885324.0  2012   

             updated_on   latitude  longitude                       location  
0  2018-02-10T03:50:01Z  41.839303 -87.639001  (41.839303195, -87.639000957)  
1  2016-02-04T06:33:39Z  41.840856 -87.656867   (41.840856037, -87.65686732)  

[2 rows x 22 columns]

In [25]:
# Check the top 10 location description
data['location_description'].value_counts().head(10)

STREET                            3692
RESIDENCE                         2695
SIDEWALK                          1202
APARTMENT                         1197
PARKING LOT/GARAGE(NON.RESID.)     454
OTHER                              438
RESIDENCE-GARAGE                   340
ALLEY                              332
SCHOOL, PUBLIC, BUILDING           302
VEHICLE NON-COMMERCIAL             235
Name: location_description, dtype: int64

In [26]:
# check the primary type
data['primary_type'].value_counts().head(10)

BATTERY                2479
THEFT                  2346
CRIMINAL DAMAGE        2010
NARCOTICS              1385
ASSAULT                 958
OTHER OFFENSE           916
BURGLARY                799
MOTOR VEHICLE THEFT     688
DECEPTIVE PRACTICE      528
ROBBERY                 479
Name: primary_type, dtype: int64

In [27]:
data.columns

Index(['unique_key', 'case_number', 'date', 'block', 'iucr', 'primary_type',
       'description', 'location_description', 'arrest', 'domestic', 'beat',
       'district', 'ward', 'community_area', 'fbi_code', 'x_coordinate',
       'y_coordinate', 'year', 'updated_on', 'latitude', 'longitude',
       'location'],
      dtype='object')

- focus on battery, theft, criminal damage
- correlation with (arrest, domestic, description), (location description, location)
- year

In [28]:
result = []
for crime_type in data['primary_type']:
    if crime_type == 'BATTERY':
        result.append(crime_type)
result[:4]

['BATTERY', 'BATTERY', 'BATTERY', 'BATTERY']

In [29]:
# Check
'CRIMINAL DAMAGE' in result, 'THEFT' in result, 'BATTERY' in result, 'NARCOTICS'in result, 'OTHER OFFENCE' in result, 

(False, False, True, False, False)

In [30]:
# Check BATTERY relation with description, arrest, domestic
battery_crime = data[data['primary_type'] == 'BATTERY' ]
battery_crime.shape

(2479, 22)

In [31]:
# take a look at description column
print(battery_crime['description'].value_counts().shape)
print(battery_crime['description'].value_counts().head(31))


(31,)
DOMESTIC BATTERY SIMPLE                                        1172
SIMPLE                                                          841
AGGRAVATED: OTHER DANG WEAPON                                   129
AGGRAVATED: HANDGUN                                              50
AGGRAVATED:KNIFE/CUTTING INSTR                                   49
AGGRAVATED DOMESTIC BATTERY: OTHER DANG WEAPON                   34
PRO EMP HANDS NO/MIN INJURY                                      33
AGGRAVATED DOMESTIC BATTERY: KNIFE/CUTTING INST                  31
AGG PO HANDS NO/MIN INJURY                                       28
AGGRAVATED - HANDGUN                                             20
AGG: HANDS/FIST/FEET NO/MINOR INJURY                             18
AGGRAVATED DOMESTIC BATTERY: HANDS/FIST/FEET SERIOUS INJURY      12
AGGRAVATED - OTHER DANGEROUS WEAPON                              11
AGGRAVATED DOMESTIC BATTERY - OTHER DANGEROUS WEAPON              8
AGGRAVATED - KNIFE / CUTTING INSTRUMENT   

Check the difference between DOMESTIC BATTERY SIMPLE and SIMPLE

In [32]:
battery_crime[battery_crime['description'] == 'DOMESTIC BATTERY SIMPLE'].tail(2)

unique_key case_number                  date                 block  \
9356     9412643    HW556314  2013-12-02T06:00:00Z   057XX W HIGGINS AVE   
9357     5937070    HN735766  2007-11-30T08:55:00Z  035XX S ARTESIAN AVE   

      iucr primary_type              description location_description  arrest  \
9356  0486      BATTERY  DOMESTIC BATTERY SIMPLE            APARTMENT   False   
9357  0486      BATTERY  DOMESTIC BATTERY SIMPLE               STREET   False   

      domestic  ...  ward  community_area  fbi_code  x_coordinate  \
9356     False  ...  45.0            11.0       08B     1136960.0   
9357      True  ...  12.0            59.0       08B     1160597.0   

     y_coordinate  year            updated_on   latitude  longitude  \
9356    1932689.0  2013  2018-02-10T03:50:01Z  41.971456 -87.771765   
9357    1881160.0  2007  2018-02-28T03:56:25Z  41.829598 -87.686277   

                           location  
9356  (41.971456136, -87.771764596)  
9357  (41.829598355, -87.686277105)  

[2 rows x 22 columns]

In [33]:
battery_crime[battery_crime['description'] == 'SIMPLE'].tail(2)

unique_key case_number                  date                block  iucr  \
8771    10079690    HY268175  2015-05-20T03:40:00Z  133XX S LANGLEY AVE  0460   
8772     5939953    HN738353  2007-12-02T11:40:00Z    107XX S EWING AVE  0460   

     primary_type description     location_description  arrest  domestic  ...  \
8771      BATTERY      SIMPLE  CHA PARKING LOT/GROUNDS    True     False  ...   
8772      BATTERY      SIMPLE       GROCERY FOOD STORE   False     False  ...   

      ward  community_area  fbi_code  x_coordinate y_coordinate  year  \
8771   9.0            54.0       08B     1183326.0    1817142.0  2015   
8772  10.0            52.0       08B     1202188.0    1834289.0  2007   

                updated_on   latitude  longitude  \
8771  2018-02-10T03:50:01Z  41.653426 -87.604874   
8772  2018-02-28T03:56:25Z  41.700021 -87.535279   

                           location  
8771  (41.653425905, -87.604873764)  
8772   (41.700020818, -87.53527894)  

[2 rows x 22 columns]

conclusion: The two are different since the fbi_code are different. No correlation with domestic column

# Question to answer

From this, it safe not to use description in analysis, because:
- description is messy
- Instead, how about battery crime that lead to arrest:
- how about its description?

In [34]:
battery_crime[battery_crime['arrest'] == True].head(2)

unique_key case_number                  date              block  iucr  \
151    12399489    JE271794  2021-06-19T12:01:00Z    008XX W 52ND ST  0497   
157     4719795    HM325669  2006-05-02T10:45:00Z  047XX S BISHOP ST  0484   

    primary_type                                        description  \
151      BATTERY  AGGRAVATED DOMESTIC BATTERY - OTHER DANGEROUS ...   
157      BATTERY                        PRO EMP HANDS NO/MIN INJURY   

         location_description  arrest  domestic  ...  ward  community_area  \
151                    STREET    True      True  ...  20.0            61.0   
157  SCHOOL, PUBLIC, BUILDING    True     False  ...  20.0            61.0   

     fbi_code  x_coordinate y_coordinate  year            updated_on  \
151       04B     1171616.0    1870374.0  2021  2021-09-20T04:45:04Z   
157       08B     1167467.0    1873258.0  2006  2018-02-28T03:56:25Z   

      latitude  longitude                       location  
151  41.799766 -87.646165  (41.799765636, -87.646165129)  
157  41.807770 -87.661298  (41.807769669, -87.661297945)  

[2 rows x 22 columns]

546 BATTERY crime happened to lead to arrests

In [35]:
# Check the word 'GUN' for battery crime description
battery_crime.loc[battery_crime['description'].str.contains("GUN")][40:42]

unique_key case_number                  date               block  iucr  \
4594    12217022    JD426533  2020-11-09T08:05:00Z     019XX S TROY ST  041A   
4595     9890965    HX541206  2014-12-13T08:10:00Z  012XX S ALBANY AVE  041A   

     primary_type           description location_description  arrest  \
4594      BATTERY  AGGRAVATED - HANDGUN             SIDEWALK   False   
4595      BATTERY   AGGRAVATED: HANDGUN                ALLEY   False   

      domestic  ...  ward  community_area  fbi_code  x_coordinate  \
4594     False  ...  24.0            29.0       04B     1155664.0   
4595     False  ...  24.0            29.0       04B     1155906.0   

     y_coordinate  year            updated_on   latitude  longitude  \
4594    1890335.0  2020  2020-12-17T03:44:58Z  41.854876 -87.704130   
4595    1894238.0  2014  2018-02-10T03:50:01Z  41.865582 -87.703136   

                           location  
4594  (41.854876172, -87.704129692)  
4595    (41.86558156, -87.70313629)  

[2 rows x 22 columns]

No Shooting, but mostly AGGRAVATED HANDGUN 

In [36]:
# Check where battery crime location in school or building
school_public_building = battery_crime.loc[battery_crime['location_description'].str.contains("SCHOOL|BUILDING|PUBLIC")]
battery_crime.loc[battery_crime['location_description'].str.contains("SCHOOL|BUILDING|PUBLIC")].tail(2)

unique_key case_number                  date            block  iucr  \
8812    10498608    HZ239990  2016-04-25T06:15:00Z  021XX E 87TH ST  0486   
8902    10086893    HY275780  2015-05-26T03:06:00Z  045XX W 46TH ST  0486   

     primary_type              description            location_description  \
8812      BATTERY  DOMESTIC BATTERY SIMPLE       SCHOOL, PRIVATE, BUILDING   
8902      BATTERY  DOMESTIC BATTERY SIMPLE  FACTORY/MANUFACTURING BUILDING   

      arrest  domestic  ...  ward  community_area  fbi_code  x_coordinate  \
8812    True      True  ...   8.0            45.0       08B     1191768.0   
8902    True      True  ...  14.0            57.0       08B     1147098.0   

     y_coordinate  year            updated_on   latitude  longitude  \
8812    1847705.0  2016  2018-02-10T03:50:01Z  41.737094 -87.572998   
8902    1873755.0  2015  2018-02-10T03:50:01Z  41.809546 -87.735994   

                           location  
8812  (41.737094305, -87.572998178)  
8902    (41.809546288, -87.7359941)  

[2 rows x 22 columns]

160 of battery crime happened in school or building or public

In [37]:
# Battery crime in school public building lead to arrests
battery_school_arrest = school_public_building[school_public_building['arrest'] == True]
battery_school_arrest[23:27]

unique_key case_number                  date                 block  \
7585     5553491    HN361985  2007-05-24T10:43:00Z  092XX S PHILLIPS AVE   
7610     7889261    HT117639  2011-01-13T10:00:00Z     131XX S CARVER DR   
7818     3801338    HL161728  2005-02-04T10:32:00Z      008XX E 103RD ST   
7948     7484723    HS278448  2010-04-27T12:05:00Z      008XX E 103RD ST   

      iucr primary_type                          description  \
7585  0484      BATTERY          PRO EMP HANDS NO/MIN INJURY   
7610  0479      BATTERY  AGG: HANDS/FIST/FEET SERIOUS INJURY   
7818  0420      BATTERY       AGGRAVATED:KNIFE/CUTTING INSTR   
7948  0484      BATTERY          PRO EMP HANDS NO/MIN INJURY   

          location_description  arrest  domestic  ...  ward  community_area  \
7585   SCHOOL, PUBLIC, GROUNDS    True     False  ...   7.0            48.0   
7610  SCHOOL, PUBLIC, BUILDING    True     False  ...   9.0            54.0   
7818  SCHOOL, PUBLIC, BUILDING    True     False  ...   9.0            50.0   
7948  SCHOOL, PUBLIC, BUILDING    True     False  ...   9.0            50.0   

      fbi_code  x_coordinate y_coordinate  year            updated_on  \
7585       08B     1194110.0    1844254.0  2007  2018-02-10T03:50:01Z   
7610       04B     1186850.0    1818609.0  2011  2016-02-04T06:33:39Z   
7818       04B     1183988.0    1836860.0  2005  2018-02-10T03:50:01Z   
7948       08B     1183988.0    1836860.0  2010  2018-02-10T03:50:01Z   

       latitude  longitude                       location  
7585  41.727567 -87.564531  (41.727567355, -87.564530885)  
7610  41.657369 -87.591934   (41.65736912, -87.591933595)  
7818  41.707519 -87.601839  (41.707519433, -87.601839023)  
7948  41.707519 -87.601839  (41.707519433, -87.601839023)  

[4 rows x 22 columns]

44 battery crime cases in school, private, building lead to arrest

In [38]:
# Battery crime in school, private, building that lead to arrest, and domestic is True
battery_school_arrest[battery_school_arrest['domestic'] == True]

unique_key case_number                  date            block  iucr  \
8812    10498608    HZ239990  2016-04-25T06:15:00Z  021XX E 87TH ST  0486   
8902    10086893    HY275780  2015-05-26T03:06:00Z  045XX W 46TH ST  0486   

     primary_type              description            location_description  \
8812      BATTERY  DOMESTIC BATTERY SIMPLE       SCHOOL, PRIVATE, BUILDING   
8902      BATTERY  DOMESTIC BATTERY SIMPLE  FACTORY/MANUFACTURING BUILDING   

      arrest  domestic  ...  ward  community_area  fbi_code  x_coordinate  \
8812    True      True  ...   8.0            45.0       08B     1191768.0   
8902    True      True  ...  14.0            57.0       08B     1147098.0   

     y_coordinate  year            updated_on   latitude  longitude  \
8812    1847705.0  2016  2018-02-10T03:50:01Z  41.737094 -87.572998   
8902    1873755.0  2015  2018-02-10T03:50:01Z  41.809546 -87.735994   

                           location  
8812  (41.737094305, -87.572998178)  
8902    (41.809546288, -87.7359941)  

[2 rows x 22 columns]

# Another Question:
- How about location of the crime spread?

In [39]:
data['location']

0        (41.839303195, -87.639000957)
1         (41.840856037, -87.65686732)
2        (41.841233568, -87.637831421)
3         (41.83204543, -87.640044437)
4         (41.84091182, -87.651912438)
                     ...              
14033    (41.722191224, -87.595504017)
14034      (41.9821052, -87.773204738)
14035    (41.733188721, -87.591403051)
14036     (41.725457345, -87.55805009)
14037    (41.923613437, -87.793511665)
Name: location, Length: 13889, dtype: object

## Export Data to Plot in Google Maps

In [40]:
# export a dataframe
export = data[['primary_type', 'location']]
export.to_csv('./location_cases.csv', 
                 index = False)

In [41]:
# check the data
map_plot = pd.read_csv('./location_cases.csv')
map_plot[map_plot['primary_type'] == 'BATTERY']

primary_type                       location
151       BATTERY  (41.799765636, -87.646165129)
157       BATTERY  (41.807769669, -87.661297945)
158       BATTERY  (41.797022081, -87.674392638)
160       BATTERY  (41.797046142, -87.661021252)
161       BATTERY  (41.802669989, -87.655090533)
...           ...                            ...
9259      BATTERY   (41.72221878, -87.592961524)
9260      BATTERY  (41.811747899, -87.632659218)
9261      BATTERY    (41.98375277, -87.77680582)
9262      BATTERY  (41.971456136, -87.771764596)
9263      BATTERY  (41.829598355, -87.686277105)

[2479 rows x 2 columns]

In [42]:
# check for null again LOL
map_plot.isna().any()

primary_type    False
location        False
dtype: bool

Note: The maps can only plot 2000 points
- So, this is a drawback

# The Map

<iframe src="https://www.google.com/maps/d/u/0/embed?mid=1AD7vQSQKL4JFIchhr6hnXEAaKX5f47g&ehbc=2E312F" width="640" height="480"></iframe>

Note: 
- I still don't know how to plot it using Tableau or others, so I was looking for a way to visualise the crime location on a map. Found that Google Maps allow to create a custom map using a dataset.
- I was trying to visualise it using heatmap or smt

# Let's explore another type of crime

### Focus on
- stalking, kidnapping, sex offence, prostitution, crim sexual assault, criminal sexual assault

In [43]:
# sex offence, crim sexual assault, stalking, kidnapping, criminal sexuak assault

In [44]:
data['primary_type'].value_counts()

BATTERY                              2479
THEFT                                2346
CRIMINAL DAMAGE                      2010
NARCOTICS                            1385
ASSAULT                               958
OTHER OFFENSE                         916
BURGLARY                              799
MOTOR VEHICLE THEFT                   688
DECEPTIVE PRACTICE                    528
ROBBERY                               479
CRIMINAL TRESPASS                     405
WEAPONS VIOLATION                     240
OFFENSE INVOLVING CHILDREN            142
PUBLIC PEACE VIOLATION                115
PROSTITUTION                          101
SEX OFFENSE                            51
INTERFERENCE WITH PUBLIC OFFICER       48
ARSON                                  37
CRIM SEXUAL ASSAULT                    35
GAMBLING                               30
HOMICIDE                               23
LIQUOR LAW VIOLATION                   20
KIDNAPPING                             18
STALKING                          

Replace 'CRIM SEXUAL ASSAULT' with 'CRIMINAL SEXUAL ASSAULT'

In [45]:
data['primary_type'].replace(['CRIM SEXUAL ASSAULT'], 'CRIMINAL SEXUAL ASSAULT', inplace=True)

C:\Users\ASUS\AppData\Local\Temp\ipykernel_4396\430319093.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['primary_type'].replace(['CRIM SEXUAL ASSAULT'], 'CRIMINAL SEXUAL ASSAULT', inplace=True)


In [46]:
# Check
data['primary_type'].value_counts()

BATTERY                              2479
THEFT                                2346
CRIMINAL DAMAGE                      2010
NARCOTICS                            1385
ASSAULT                               958
OTHER OFFENSE                         916
BURGLARY                              799
MOTOR VEHICLE THEFT                   688
DECEPTIVE PRACTICE                    528
ROBBERY                               479
CRIMINAL TRESPASS                     405
WEAPONS VIOLATION                     240
OFFENSE INVOLVING CHILDREN            142
PUBLIC PEACE VIOLATION                115
PROSTITUTION                          101
SEX OFFENSE                            51
INTERFERENCE WITH PUBLIC OFFICER       48
CRIMINAL SEXUAL ASSAULT                43
ARSON                                  37
GAMBLING                               30
HOMICIDE                               23
LIQUOR LAW VIOLATION                   20
KIDNAPPING                             18
STALKING                          

In [47]:
# Check LOL
35+8
# Can't believe I do this 

43

In [48]:
# Check by looking for the spesific name ('CRIM SEXUAL ASSAULT)
data[data['primary_type'] == 'CRIM SEXUAL ASSAULT']

Empty DataFrame
Columns: [unique_key, case_number, date, block, iucr, primary_type, description, location_description, arrest, domestic, beat, district, ward, community_area, fbi_code, x_coordinate, y_coordinate, year, updated_on, latitude, longitude, location]
Index: []

[0 rows x 22 columns]

### Let's back to the question
- stalking, kidnapping, sex offence, prostitution, criminal sexual assault

### Let's point out this
- I want to see if there's a stalking crime that lead to kidnapping, sex offence, or criminal sexual assault (same case)

In [49]:
# I want to look at the date first to make sure
# First is to look at the stalking crimes
stalking = data[data['primary_type'] == 'STALKING']
stalking[:2]

unique_key case_number                  date            block  iucr  \
479     7499228    HS302309  2010-05-11T02:20:00Z  043XX S WOOD ST  0580   
502    10907549    JA220167  2017-03-30T01:00:00Z  005XX W 41ST ST  0580   

    primary_type description location_description  arrest  domestic  ...  \
479     STALKING      SIMPLE               STREET   False      True  ...   
502     STALKING      SIMPLE            RESIDENCE   False      True  ...   

     ward  community_area  fbi_code  x_coordinate y_coordinate  year  \
479  12.0            61.0       08A     1165073.0    1875957.0  2010   
502  11.0            61.0       08A     1173413.0    1877729.0  2017   

               updated_on   latitude  longitude                       location  
479  2018-02-10T03:50:01Z  41.815227 -87.670002  (41.815227068, -87.670002097)  
502  2018-02-10T03:50:01Z  41.819909 -87.639357  (41.819908866, -87.639357255)  

[2 rows x 22 columns]

In [50]:
# now let's check kidnapping crimes
kidnapping = data[data['primary_type'] == 'KIDNAPPING']
kidnapping[:2]
# Kinda hard since we can only look at the case number, date, and the location

unique_key case_number                  date                block  iucr  \
181    10515771    HZ257735  2016-04-01T03:00:00Z      019XX W 50TH ST  1792   
264     3434252    HK500619  2004-07-16T09:10:00Z  049XX S LOOMIS BLVD  1792   

    primary_type               description     location_description  arrest  \
181   KIDNAPPING  CHILD ABDUCTION/STRANGER  SCHOOL, PUBLIC, GROUNDS   False   
264   KIDNAPPING  CHILD ABDUCTION/STRANGER                 SIDEWALK   False   

     domestic  ...  ward  community_area  fbi_code  x_coordinate y_coordinate  \
181     False  ...  16.0            61.0        20     1164477.0    1871523.0   
264     False  ...  16.0            61.0        20     1167839.0    1871794.0   

     year            updated_on   latitude  longitude  \
181  2016  2018-02-10T03:50:01Z  41.803072 -87.672313   
264  2004  2018-02-28T03:56:25Z  41.803744 -87.659976   

                          location  
181  (41.803072258, -87.672313423)  
264    (41.8037443, -87.659975618)  

[2 rows x 22 columns]

In [51]:
sex_offence = data[data['primary_type'] == 'SEX OFFENSE']
sex_offence[:2]

unique_key case_number                  date              block  iucr  \
147     7023050    HR429893  2009-07-13T05:00:00Z    009XX W 51ST PL  1563   
184    10872798    JA172158  2017-03-09T08:30:00Z  049XX S LAFLIN ST  1570   

    primary_type            description location_description  arrest  \
147  SEX OFFENSE  CRIMINAL SEXUAL ABUSE            APARTMENT   False   
184  SEX OFFENSE       PUBLIC INDECENCY             SIDEWALK   False   

     domestic  ...  ward  community_area  fbi_code  x_coordinate y_coordinate  \
147     False  ...  20.0            61.0        17     1170888.0    1870681.0   
184     False  ...  20.0            61.0        17     1167174.0    1871825.0   

     year            updated_on   latitude  longitude  \
147  2009  2018-02-28T03:56:25Z  41.800624 -87.648826   
184  2017  2018-02-10T03:50:01Z  41.803844 -87.662414   

                          location  
147  (41.800624017, -87.648825932)  
184  (41.803843637, -87.662413619)  

[2 rows x 22 columns]

In [52]:
crim_sex_assault = data[data['primary_type'] == 'CRIMINAL SEXUAL ASSAULT']
crim_sex_assault[:2]

unique_key case_number                  date                   block  \
193     5835160    HN639009  2007-10-08T07:30:00Z         047XX S WOOD ST   
262     2720513    HJ349768  2003-05-04T10:00:00Z  050XX S MARSHFIELD AVE   

     iucr             primary_type     description location_description  \
193  0281  CRIMINAL SEXUAL ASSAULT  NON-AGGRAVATED            APARTMENT   
262  0266  CRIMINAL SEXUAL ASSAULT       PREDATORY            RESIDENCE   

     arrest  domestic  ...  ward  community_area  fbi_code  x_coordinate  \
193   False     False  ...  20.0            61.0        02     1165145.0   
262    True     False  ...  16.0            61.0        02     1166193.0   

    y_coordinate  year            updated_on   latitude  longitude  \
193    1873294.0  2007  2018-02-28T03:56:25Z  41.807918 -87.669813   
262    1871262.0  2003  2018-02-28T03:56:25Z  41.802320 -87.666027   

                          location  
193   (41.807917966, -87.66981341)  
262  (41.802319657, -87.666027478)  

[2 rows x 22 columns]

In [62]:
# stalking, sex_offence, kidnapping, crim_sex_assault

In [ ]:
# Check the block, if any match value 
ct1 = 0
for so_crime_location in sex_offence['block']:
    # print(crime_location)
    ct1 +=1
    print(ct1, stalking['block'] == so_crime_location)
print(ct1)
# ANS: No

In [55]:
sex_offence.shape

(51, 22)

In [ ]:
# Check the block, if any match value
ct2 = 0
for ki_crime_location in kidnapping['block']:
    # print(crime_location)
    ct2 +=1
    print(ct2, stalking['block'] == ki_crime_location)
print(ct2)
# ANS: No

In [57]:
kidnapping.shape

(18, 22)

In [ ]:
# Check the block, if any match value
ct3 = 0
for cr_crime_location in crim_sex_assault['block']:
    # print(crime_location)
    ct3 +=1
    print(ct3, stalking['block'] == cr_crime_location)
print(ct3)
# ANS: NO

In [59]:
crim_sex_assault.shape

(43, 22)

Can't find any case from stalking crime that has relation on sex offence, kidnapping, or criminal sexual assault

# Let's move to other Questions

Check whether any crime description were involved with child
- I actually saw child in crime description when looking at the kidnapping crime
- That's how I got into this question

In [64]:
data.loc[data['description'].str.contains("CHILD")][73:80]

unique_key case_number                  date                   block  \
7249    10044053    HY232782  2015-04-14T01:00:00Z        038XX W 111TH ST   
7259     4456628    HL753512  2005-11-24T09:15:00Z  035XX S WINCHESTER AVE   
7261     3172251    HK173739  2002-11-01T12:01:00Z   037XX S HERMITAGE AVE   
7265    11937091    JC565609  2019-12-20T05:00:00Z    051XX N NORMANDY AVE   
7309     2594504    HJ182886  2003-02-14T05:30:00Z     027XX N NORDICA AVE   
7351     7814308    HS624184  2010-11-18T03:00:00Z        100XX S AVENUE L   
7377     2610355    HJ209096  2003-02-27T10:00:00Z    105XX S MARYLAND AVE   

      iucr                primary_type                     description  \
7249  1582  OFFENSE INVOLVING CHILDREN               CHILD PORNOGRAPHY   
7259  1790  OFFENSE INVOLVING CHILDREN                 CHILD ABDUCTION   
7261  1754  OFFENSE INVOLVING CHILDREN  AGG SEX ASSLT OF CHILD FAM MBR   
7265  1750  OFFENSE INVOLVING CHILDREN                     CHILD ABUSE   
7309  1750  OFFENSE INVOLVING CHILDREN                     CHILD ABUSE   
7351  1750  OFFENSE INVOLVING CHILDREN                     CHILD ABUSE   
7377  1790  OFFENSE INVOLVING CHILDREN                 CHILD ABDUCTION   

          location_description  arrest  domestic  ...  ward  community_area  \
7249  SCHOOL, PUBLIC, BUILDING    True     False  ...  19.0            74.0   
7259                 RESIDENCE   False      True  ...  11.0            59.0   
7261                 RESIDENCE    True     False  ...  11.0            59.0   
7265                 RESIDENCE   False     False  ...  45.0            10.0   
7309                 RESIDENCE    True      True  ...  36.0            18.0   
7351                 APARTMENT   False      True  ...  10.0            52.0   
7377                 RESIDENCE   False      True  ...   9.0            50.0   

      fbi_code  x_coordinate y_coordinate  year            updated_on  \
7249        17     1152573.0    1830709.0  2015  2018-02-10T03:50:01Z   
7259        20     1163934.0    1881136.0  2005  2018-02-28T03:56:25Z   
7261        02     1165301.0    1879815.0  2002  2018-02-28T03:56:25Z   
7265       08B     1130918.0    1933740.0  2019  2020-01-03T03:56:42Z   
7309        20     1128624.0    1917323.0  2003  2018-02-28T03:56:25Z   
7351        20     1201813.0    1839112.0  2010  2018-02-10T03:50:01Z   
7377        20     1183775.0    1835225.0  2003  2018-02-28T03:56:25Z   

       latitude  longitude                       location  
7249  41.691315 -87.717043  (41.691314623, -87.717043381)  
7259  41.829463 -87.674034  (41.829462872, -87.674034362)  
7261  41.825809 -87.669056   (41.82580901, -87.669056347)  
7265  41.974447 -87.793958  (41.974446713, -87.793957866)  
7309  41.929436 -87.802767  (41.929435928, -87.802767497)  
7351  41.713265 -87.536489  (41.713265058, -87.536488591)  
7377  41.703038 -87.602670  (41.703037749, -87.602669826)  

[7 rows x 22 columns]

There's 98 cases that involved child or children. Let's explore this

In [68]:
# Lets count distinct the crime description
# To show what kinda crime involved with child
crime_child = data.loc[data['description'].str.contains("CHILD")]
crime_child['description'].value_counts()

CHILD ABUSE                                            45
ENDANGERING LIFE/HEALTH CHILD                          16
CHILD ABDUCTION                                        11
AGG SEX ASSLT OF CHILD FAM MBR                         10
CHILD ABDUCTION/STRANGER                                8
AGGRAVATED SEXUAL ASSAULT OF CHILD BY FAMILY MEMBER     3
CHILD PORNOGRAPHY                                       2
CHILD ABANDONMENT                                       1
SEX ASSLT OF CHILD BY FAM MBR                           1
SEXUAL EXPLOITATION OF A CHILD                          1
Name: description, dtype: int64

Clean the data:
- Combine: AGG SEX ASSLT OF CHILD FAM MBR to AGGRAVATED SEXUAL ASSAULT OF CHILD BY FAMILY MEMBER
- Change SEX ASSLT OF CHILD BY FAM MBR to SEXUAL ASSAULT OF CHILD BY FAMILY MEMBER*
- I decide to differentiate aggravated from the similar crime

In [71]:
# clean the data
data['description'].replace(['AGG SEX ASSLT OF CHILD FAM MBR'], 'AGGRAVATED SEXUAL ASSAULT OF CHILD BY FAMILY MEMBER', inplace=True)
data['description'].replace(['SEX ASSLT OF CHILD BY FAM MBR'], 'SEXUAL ASSAULT OF CHILD BY FAMILY MEMBER', inplace=True)

C:\Users\ASUS\AppData\Local\Temp\ipykernel_4396\591902042.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['description'].replace(['AGG SEX ASSLT OF CHILD FAM MBR'], 'AGGRAVATED SEXUAL ASSAULT OF CHILD BY FAMILY MEMBER', inplace=True)
C:\Users\ASUS\AppData\Local\Temp\ipykernel_4396\591902042.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['description'].replace(['SEX ASSLT OF CHILD BY FAM MBR'], 'SEXUAL ASSAULT OF CHILD BY FAMILY MEMBER', inplace=True)


SIMPLE                                         1481
DOMESTIC BATTERY SIMPLE                        1172
TO VEHICLE                                     1084
$500 AND UNDER                                  960
TO PROPERTY                                     791
                                               ... 
SEX OFFENDER - FAIL TO REGISTER NEW ADDRESS       1
ARMED WHILE UNDER THE INFLUENCE                   1
EXTORTION                                         1
AGGRAVATED - OTHER                                1
ATTEMPT - AUTOMOBILE                              1
Name: description, Length: 272, dtype: int64

In [73]:
# check the data if the above command has applied
crime_child = data.loc[data['description'].str.contains("CHILD")]
crime_child['description'].value_counts()

CHILD ABUSE                                            45
ENDANGERING LIFE/HEALTH CHILD                          16
AGGRAVATED SEXUAL ASSAULT OF CHILD BY FAMILY MEMBER    13
CHILD ABDUCTION                                        11
CHILD ABDUCTION/STRANGER                                8
CHILD PORNOGRAPHY                                       2
CHILD ABANDONMENT                                       1
SEXUAL ASSAULT OF CHILD BY FAMILY MEMBER                1
SEXUAL EXPLOITATION OF A CHILD                          1
Name: description, dtype: int64

Now that the description looks clean, we can continue exploring

In [76]:
# let's see th location first
crime_child['location_description'].value_counts()

RESIDENCE                         49
APARTMENT                         28
SIDEWALK                           5
STREET                             3
CHA APARTMENT                      3
SCHOOL, PUBLIC, GROUNDS            2
OTHER                              2
PARKING LOT/GARAGE(NON.RESID.)     1
SCHOOL, PRIVATE, GROUNDS           1
ALLEY                              1
SCHOOL, PUBLIC, BUILDING           1
DAY CARE CENTER                    1
SCHOOL, PRIVATE, BUILDING          1
Name: location_description, dtype: int64

In [80]:
# let's check the cases where the assailant is arrested
crime_child[crime_child['arrest'] == True].shape

(25, 22)

There's 25 cases where the assailant of the crime got arrested

In [83]:
# let's check this cases description: ENDANGERING LIFE/HEALTH CHILD
endanger_child_life = crime_child[crime_child['description'] == 'ENDANGERING LIFE/HEALTH CHILD']

In [84]:
# 16 cases of endangering lefe/health child
# Let's check how many of them were arrested
endanger_child_life[endanger_child_life['arrest'] == True]


unique_key case_number                  date               block  iucr  \
229      6406388    HP487881  2008-08-01T08:36:00Z     011XX W 51ST ST  1710   
286      6660138    HP732291  2008-12-13T04:13:00Z  047XX S PAULINA ST  1710   
7035     3015723    HJ719942  2003-10-26T06:00:00Z   042XX S ELLIS AVE  1710   
7925     8013294    HT244958  2011-04-11T02:53:00Z     017XX W 35TH ST  1710   
7977     6987764    HR392422  2009-06-24T11:00:00Z    101XX S AVENUE L  1710   

                    primary_type                    description  \
229   OFFENSE INVOLVING CHILDREN  ENDANGERING LIFE/HEALTH CHILD   
286   OFFENSE INVOLVING CHILDREN  ENDANGERING LIFE/HEALTH CHILD   
7035  OFFENSE INVOLVING CHILDREN  ENDANGERING LIFE/HEALTH CHILD   
7925  OFFENSE INVOLVING CHILDREN  ENDANGERING LIFE/HEALTH CHILD   
7977  OFFENSE INVOLVING CHILDREN  ENDANGERING LIFE/HEALTH CHILD   

                location_description  arrest  domestic  ...  ward  \
229   PARKING LOT/GARAGE(NON.RESID.)    True     False  ...  16.0   
286                        RESIDENCE    True      True  ...  20.0   
7035                       APARTMENT    True     False  ...   4.0   
7925       SCHOOL, PRIVATE, BUILDING    True     False  ...  11.0   
7977                       RESIDENCE    True      True  ...  10.0   

      community_area  fbi_code  x_coordinate y_coordinate  year  \
229             61.0        26     1169429.0    1870975.0  2008   
286             61.0        26     1165810.0    1873260.0  2008   
7035            36.0        26     1183638.0    1876974.0  2003   
7925            59.0        26     1165374.0    1881513.0  2011   
7977            52.0        26     1201820.0    1838320.0  2009   

                updated_on   latitude  longitude  \
229   2018-02-28T03:56:25Z  41.801463 -87.654168   
286   2018-02-28T03:56:25Z  41.807811 -87.667375   
7035  2018-02-28T03:56:25Z  41.817604 -87.601872   
7925  2018-02-10T03:50:01Z  41.830467 -87.668740   
7977  2018-02-28T03:56:25Z  41.711092 -87.536490   

                           location  
229    (41.801462544, -87.65416803)  
286   (41.807810552, -87.667375326)  
7035   (41.81760435, -87.601871584)  
7925  (41.830466952, -87.668740336)  
7977  (41.711091569, -87.536489798)  

[5 rows x 22 columns]

# Another Question

### What's the crime types in the last 3 years?

In [94]:
recent_data = data.loc[data['date'].str.contains("2020|2021|2022")]
recent_data.head()

unique_key case_number                  date               block  iucr  \
4     12689202    JF230287  2022-05-04T05:05:00Z   029XX S THROOP ST  1310   
18    12579030    JE491061  2021-12-27T11:55:00Z   034XX S MORGAN ST  1320   
33    12270763    JE114147  2021-01-16T01:44:00Z  025XX S HALSTED ST  1320   
42    12478340    JE368504  2021-05-05T12:01:00Z     030XX S LOCK ST  1320   
44    12260562    JE101542  2021-01-01T07:00:00Z  027XX S WALLACE ST  1320   

       primary_type  description location_description  arrest  domestic  ...  \
4   CRIMINAL DAMAGE  TO PROPERTY            RESIDENCE   False     False  ...   
18  CRIMINAL DAMAGE   TO VEHICLE               STREET   False     False  ...   
33  CRIMINAL DAMAGE   TO VEHICLE               STREET   False     False  ...   
42  CRIMINAL DAMAGE   TO VEHICLE                ALLEY   False     False  ...   
44  CRIMINAL DAMAGE   TO VEHICLE               STREET   False     False  ...   

    ward  community_area  fbi_code  x_coordinate y_coordinate  year  \
4   11.0            60.0        14     1169930.0    1885355.0  2022   
18  11.0            60.0        14     1170204.0    1882170.0  2021   
33  11.0            60.0        14     1171397.0    1887419.0  2021   
42  11.0            60.0        14     1167681.0    1884595.0  2021   
44  11.0            60.0        14     1172742.0    1886595.0  2021   

              updated_on   latitude  longitude                       location  
4   2022-05-11T04:49:57Z  41.840912 -87.651912   (41.84091182, -87.651912438)  
18  2022-01-03T03:52:37Z  41.832166 -87.651000  (41.832165918, -87.650999807)  
33  2021-01-23T03:54:52Z  41.846544 -87.646469   (41.846543549, -87.64646854)  
42  2021-09-14T04:46:28Z  41.838875 -87.660187  (41.838874974, -87.660187273)  
44  2021-01-16T03:49:23Z  41.844253 -87.641557    (41.8442528, -87.641556841)  

[5 rows x 22 columns]

In [95]:
recent_data.shape

(1071, 22)

In [98]:
# check 2020
twenty_data = recent_data.loc[recent_data['date'].str.contains("2020")]
twenty_data.head()

unique_key case_number                  date                  block  \
49     12121461    JD315640  2020-07-30T08:30:00Z    031XX S STEWART AVE   
142    12120281    JD314106  2020-07-01T12:00:00Z     030XX S WALLACE ST   
226    12243438    JD456623  2020-12-11T09:22:00Z   050XX S CARPENTER ST   
230    12029076    JD207884  2020-04-13T12:50:00Z  046XX S HERMITAGE AVE   
237    12175101    JD378125  2020-09-24T06:42:00Z   043XX S WESTERN BLVD   

     iucr         primary_type                    description  \
49   1320      CRIMINAL DAMAGE                     TO VEHICLE   
142  5002        OTHER OFFENSE          OTHER VEHICLE OFFENSE   
226  0930  MOTOR VEHICLE THEFT  THEFT / RECOVERY - AUTOMOBILE   
230  1020                ARSON                        BY FIRE   
237  1477    WEAPONS VIOLATION     RECKLESS FIREARM DISCHARGE   

    location_description  arrest  domestic  ...  ward  community_area  \
49                STREET   False     False  ...  11.0            60.0   
142            RESIDENCE   False      True  ...  11.0            60.0   
226               STREET   False     False  ...  20.0            61.0   
230               STREET   False     False  ...  15.0            61.0   
237               STREET   False     False  ...  15.0            61.0   

     fbi_code  x_coordinate y_coordinate  year            updated_on  \
49         14     1174288.0    1884077.0  2020  2020-08-06T03:46:43Z   
142        26     1172789.0    1884725.0  2020  2020-08-01T03:46:24Z   
226        07     1170169.0    1871391.0  2020  2020-12-18T03:51:11Z   
230        09     1165460.0    1873875.0  2020  2020-05-14T03:50:16Z   
237        15     1161242.0    1875927.0  2020  2020-10-01T03:50:30Z   

      latitude  longitude                       location  
49   41.837309 -87.635958  (41.837308892, -87.635958365)  
142  41.839120 -87.641440  (41.839120324, -87.641439644)  
226  41.802588 -87.651442  (41.802588016, -87.651442076)  
230  41.809506 -87.668642  (41.809505617, -87.668641589)  
237  41.815225 -87.684056  (41.815225035, -87.684055692)  

[5 rows x 22 columns]

In [99]:
# check 2021
twentyone_data = recent_data.loc[recent_data['date'].str.contains("2021")]
twentyone_data.head()

unique_key case_number                  date               block  iucr  \
18    12579030    JE491061  2021-12-27T11:55:00Z   034XX S MORGAN ST  1320   
33    12270763    JE114147  2021-01-16T01:44:00Z  025XX S HALSTED ST  1320   
42    12478340    JE368504  2021-05-05T12:01:00Z     030XX S LOCK ST  1320   
44    12260562    JE101542  2021-01-01T07:00:00Z  027XX S WALLACE ST  1320   
80    12551673    JE458154  2021-11-27T09:15:00Z  030XX S HALSTED ST  1330   

         primary_type description location_description  arrest  domestic  ...  \
18    CRIMINAL DAMAGE  TO VEHICLE               STREET   False     False  ...   
33    CRIMINAL DAMAGE  TO VEHICLE               STREET   False     False  ...   
42    CRIMINAL DAMAGE  TO VEHICLE                ALLEY   False     False  ...   
44    CRIMINAL DAMAGE  TO VEHICLE               STREET   False     False  ...   
80  CRIMINAL TRESPASS     TO LAND           RESTAURANT    True     False  ...   

    ward  community_area  fbi_code  x_coordinate y_coordinate  year  \
18  11.0            60.0        14     1170204.0    1882170.0  2021   
33  11.0            60.0        14     1171397.0    1887419.0  2021   
42  11.0            60.0        14     1167681.0    1884595.0  2021   
44  11.0            60.0        14     1172742.0    1886595.0  2021   
80  11.0            60.0        26     1171462.0    1884745.0  2021   

              updated_on   latitude  longitude                       location  
18  2022-01-03T03:52:37Z  41.832166 -87.651000  (41.832165918, -87.650999807)  
33  2021-01-23T03:54:52Z  41.846544 -87.646469   (41.846543549, -87.64646854)  
42  2021-09-14T04:46:28Z  41.838875 -87.660187  (41.838874974, -87.660187273)  
44  2021-01-16T03:49:23Z  41.844253 -87.641557    (41.8442528, -87.641556841)  
80  2021-12-04T03:47:32Z  41.839204 -87.646308  (41.839204436, -87.646308495)  

[5 rows x 22 columns]

In [100]:
# check 2022
twentytwo_data = recent_data.loc[recent_data['date'].str.contains("2022")]
twentytwo_data.head()

unique_key case_number                  date                     block  \
4      12689202    JF230287  2022-05-04T05:05:00Z         029XX S THROOP ST   
61     12739545    JF291204  2022-06-23T04:32:00Z       033XX S ABERDEEN ST   
86     12735393    JF286285  2022-06-19T07:45:00Z  024XX S DAN RYAN EXPY IB   
136    12623976    JF151863  2022-02-21T09:00:00Z        030XX S NORMAL AVE   
261    12631711    JF161312  2022-03-02T12:00:00Z     046XX S HERMITAGE AVE   

     iucr                primary_type                    description  \
4    1310             CRIMINAL DAMAGE                    TO PROPERTY   
61   1320             CRIMINAL DAMAGE                     TO VEHICLE   
86   143A           WEAPONS VIOLATION  UNLAWFUL POSSESSION - HANDGUN   
136  2825               OTHER OFFENSE        HARASSMENT BY TELEPHONE   
261  1750  OFFENSE INVOLVING CHILDREN                    CHILD ABUSE   

     location_description  arrest  domestic  ...  ward  community_area  \
4               RESIDENCE   False     False  ...  11.0            60.0   
61                 STREET   False     False  ...  11.0            60.0   
86   HIGHWAY / EXPRESSWAY    True     False  ...  11.0            60.0   
136             APARTMENT   False     False  ...  11.0            60.0   
261             APARTMENT   False     False  ...  15.0            61.0   

     fbi_code  x_coordinate y_coordinate  year            updated_on  \
4          14     1169930.0    1885355.0  2022  2022-05-11T04:49:57Z   
61         14     1169527.0    1882651.0  2022  2022-06-30T04:48:57Z   
86         15     1172054.0    1888426.0  2022  2022-06-26T04:48:19Z   
136        26     1173455.0    1884722.0  2022  2022-02-28T03:47:25Z   
261       08B     1165456.0    1874018.0  2022  2022-03-09T03:48:55Z   

      latitude  longitude                       location  
4    41.840912 -87.651912   (41.84091182, -87.651912438)  
61   41.833501 -87.653470  (41.833500558, -87.653469832)  
86   41.849292 -87.644028  (41.849292395, -87.644027731)  
136  41.839097 -87.638996  (41.839097344, -87.638995843)  
261  41.809898 -87.668652  (41.809898111, -87.668652204)  

[5 rows x 22 columns]

In [101]:
# check the total cases
twenty_data.shape, twentyone_data.shape, twentytwo_data.shape

((408, 22), (438, 22), (225, 22))

In [102]:
# 2020 crime type
twenty_data['primary_type'].value_counts()

BATTERY                              73
THEFT                                67
CRIMINAL DAMAGE                      61
DECEPTIVE PRACTICE                   39
ASSAULT                              31
WEAPONS VIOLATION                    23
ROBBERY                              22
OTHER OFFENSE                        20
NARCOTICS                            17
MOTOR VEHICLE THEFT                  16
CRIMINAL TRESPASS                    10
BURGLARY                              9
OFFENSE INVOLVING CHILDREN            5
ARSON                                 3
PUBLIC PEACE VIOLATION                3
INTERFERENCE WITH PUBLIC OFFICER      3
SEX OFFENSE                           2
STALKING                              2
CONCEALED CARRY LICENSE VIOLATION     1
HOMICIDE                              1
Name: primary_type, dtype: int64

In [103]:
# 2021 crime type
twentyone_data['primary_type'].value_counts()

BATTERY                       99
THEFT                         76
CRIMINAL DAMAGE               53
ASSAULT                       42
DECEPTIVE PRACTICE            40
OTHER OFFENSE                 29
WEAPONS VIOLATION             27
ROBBERY                       15
MOTOR VEHICLE THEFT           14
NARCOTICS                     11
BURGLARY                      11
OFFENSE INVOLVING CHILDREN     6
CRIMINAL SEXUAL ASSAULT        5
CRIMINAL TRESPASS              4
HOMICIDE                       2
ARSON                          2
SEX OFFENSE                    1
STALKING                       1
Name: primary_type, dtype: int64

In [104]:
# 2022 crime type
twentytwo_data['primary_type'].value_counts()

BATTERY                       38
CRIMINAL DAMAGE               37
THEFT                         37
ASSAULT                       25
WEAPONS VIOLATION             16
OTHER OFFENSE                 13
DECEPTIVE PRACTICE            13
CRIMINAL TRESPASS             10
ROBBERY                        9
MOTOR VEHICLE THEFT            9
BURGLARY                       5
OFFENSE INVOLVING CHILDREN     4
NARCOTICS                      2
PUBLIC PEACE VIOLATION         2
CRIMINAL SEXUAL ASSAULT        2
ARSON                          2
HOMICIDE                       1
Name: primary_type, dtype: int64

I want to see the timestamp for each crime type
- from the oldest year to recent year